driver/src/main/scala/com/stratio/sparta/driver/util/HdfsUtils.scala

```diff
@@ -77,15 +91,32 @@ object HdfsUtils extends SLF4JLogging {
       }
     )
 
-    log.debug(s"Configuring HDFS with master: ${conf.get(DefaultFSProperty)} and user: $user")
-    val defaultUri = FileSystem.getDefaultUri(conf)
+    val ugi =
+      if(principalNameOption.isDefined && keytabPathOption.isDefined) {
+        val principalName = principalNameOption.getOrElse(
+          throw new IllegalStateException("principalName can not be null"))
+        val keytabPath = keytabPathOption.getOrElse(
+          throw new IllegalStateException("keytabPathOption can not be null"))
 
-    new HdfsUtils(FileSystem.get(defaultUri, conf, user), user)
+        UserGroupInformation.setConfiguration(conf)
+        Option(UserGroupInformation.loginUserFromKeytabAndReturnUGI(principalName, keytabPath))
+      } else None
 
+    new HdfsUtils(FileSystem.get(conf), user, ugi)
   }
 
   def apply(config: Option[Config]): HdfsUtils = {
     val user = config.map(_.getString("hadoopUserName")).getOrElse("stratio")
-    apply(user, hdfsConfiguration(config))
+    val principalName: Option[String] =
+      Try(Option(config.get.getString("principalName"))).getOrElse(None) match {
```

Comment:
With map, if the property hadoopUserName or principalName are not defined it will fail